pkgs/applications/networking/geph/cargo.patch

```diff
@@ -0,0 +1,26 @@
+diff --git a/Cargo.lock b/Cargo.lock
+index 64d748d..32bfb30 100644
+--- a/Cargo.lock
++++ b/Cargo.lock
+@@ -793,8 +793,7 @@ checksum = "245097e9a4535ee1e3e3931fcfcd55a796a44c643e8596ff6566d68f09b87bbc"
+ [[package]]
+ name = "conv"
+ version = "0.3.3"
+-source = "registry+https://github.com/rust-lang/crates.io-index"
+-checksum = "78ff10625fd0ac447827aa30ea8b861fead473bb60aeb73af6c1c58caf0d1299"
++source = "git+https://github.com/DanielKeep/rust-conv.git?rev=9b7e41a5e1c0508ed58cf41066744d706bba10ed#9b7e41a5e1c0508ed58cf41066744d706bba10ed"
+ dependencies = [
+  "custom_derive",
+ ]
+diff --git a/Cargo.toml b/Cargo.toml
+index 27e6f3c..bf26c97 100644
+--- a/Cargo.toml
++++ b/Cargo.toml
+@@ -108,3 +108,6 @@ opt-level=3
+ debug=0
+ lto=true
+ 
++[patch.crates-io.conv]
++git = 'https://github.com/DanielKeep/rust-conv.git'
++rev = '9b7e41a5e1c0508ed58cf41066744d706bba10ed'
```

Comment:
And why is this patch needed, adding random patches harms maintainability in the long term.

Reply:
I don't really know. There is a dep chain: geph4 -> geph4-aioutils -> dnsclient -> dnssector -> chomp -> conv. Both chomp and conv are not maintained for years. chomp requires conv 0.3.3 with std feature however conv 0.3.3 does not has that feature. I don't know why it builds fine with cargo build but can't resolve the dep with nix-build. So I have to patch it.

Reply:
Maybe cargo build recreates the lockfile with correct dependency versions? In that case I would suspect this as a upstream issue worth reporting.